### ISSUE TITLE
Race condition when reserving help channels

### ISSUE THREAD

**Issue by DanaValerie** [Java-Discord/JavaBot#360]
**Is your feature request related to a problem? Please describe.**
There appears to be a race condition when reserving help channels.
I spent a very long time typing up a detailed explanation, attaching screenshots, etc... then by the time I hit "Send", someone else had typed "hi" in the same channel or something
and it was owned by them

**Describe the solution you'd like**
It would be very useful if you could reserve an available channel with /reserve or something... This teleports you to a dormant channel that you are now the only person who has access to. Then after your first post to that channel, it then becomes "active" and moves to "Occupied"... And between the time you /reserve, and the time you type your question, you are the only person with permissions to send to the channel 
In other words, don't go to an available channel and start typing; instead type /reserve to get a dormant channel assigned to you, then it would let you start typing, and once your first message is sent, it then moves to the (OCCUPIED) section... 
Then instead of telling people to "go find an available channel", we can just tell them to type "/reserve" from anywhere
(it would also be nice if the channel name had the channel owner's name in it's title, but i'm sure you've already thought of that and there's some technical reason it can't be done)
**Describe alternatives you've considered**

**Additional context**
None.

Thank you.

**Comment by danthe1st** [comment]
If that happens, you can just copy your message to another channel instead of sending it.

Instead, you could also just type a placeholder measage in the channel for reserving and either edit the message or send a new message with the new content.

Adding a `/reserve` command has the following problems:
- It requires users to use a command in order to reserve the channel. Some people already aren't able to use `/unreserve` correctly on their first attempt which they are explicitly informed about when reserving a channel.
- The first message is pinned to the channel. This also helps in identifying who reserved the channel (if they don't delete their initial message)

Aside from that, that problem might fix itself once we get forum channels if we decide to use them due to there not being such a limitation.

**Comment by ByP4ss** [comment]
Valid points, I would like to add though:
Adding a `/reserve` command does not exclude having reserve-by-message, it could just be an additional option.

**Comment by liquiddevelopmentnet** [comment]
> Valid points, I would like to add though:
> Adding a `/reserve` command does not exclude having reserve-by-message, it could just be an additional option.

People would be definitely confused how they should reserve a channel now and what the differences are. Of course we could document it in the how to get help channel, but imo there is already too much stuff in it.

Regarding the first message, i agree with @danthe1st .

**Comment by danthe1st** [comment]
> Valid points, I would like to add though: Adding a `/reserve` command does not exclude having reserve-by-message, it could just be an additional option.

Yes but that would still result in there not being an initial message to pin and having the original message pinned is useful (for both getting an overview what the question is about/for checking who reserved the channel/in order to check the original code if there were more messages afterwards/...)

**Comment by DanaValerie** [comment]
> Yes but that would still result in there not being an initial message to pin [...]
 This is not what I was suggesting. My original suggestion was to type /reserve to be given special access to a private channel, you could then take your time describing your issue as your "pinned" message. As soon as you type that first message, it would then move to occupied like it does now.

**Comment by DevTenga** [comment]
Hello.

With the new migration to discord forum channels, this issue has been resolved.

@danthe1st , or any other contributors may kindly close this.

**Comment by danthe1st** [comment]
Indeed, the change to a forum-based help system makes this obsolete.
